libs/wingc/src/lsp/signature.rs

```diff
@@ -32,41 +34,70 @@ pub fn on_signature_help(params: lsp_types::SignatureHelpParams) -> Option<Signa
 
 			let mut scope_visitor = ScopeVisitor::new(&types, params.text_document_position_params.position);
 			scope_visitor.visit_scope(root_scope);
-			let expr = scope_visitor.call_expr?;
-			let env = scope_visitor.call_env?;
 
 			let sig_data: (
 				crate::type_check::UnsafeRef<crate::type_check::Type>,
 				&crate::ast::ArgList,
-			) = match &expr.kind {
-				ExprKind::New(new_expr) => {
-					let New { class, arg_list, .. } = new_expr;
-
-					let Some(t) = resolve_user_defined_type(class, &types.get_scope_env(&root_scope), 0).ok() else {
-						return None;
-					};
-
-					let init_lookup = t.as_class()?.env.lookup(
-						&Symbol {
-							name: CLASS_INIT_NAME.into(),
-							span: Default::default(),
-						},
-						None,
-					);
+			) = if scope_visitor.call_stmt.is_some() {
+				let stmt = scope_visitor.call_stmt?;
+				let class = scope_visitor.class?;
+
+				match &stmt.kind {
+					StmtKind::SuperConstructor { arg_list } => {
+						if let Some(p) = &class.parent {
+							let t = resolve_user_defined_type(&p, &types.get_scope_env(&root_scope), 0).ok()?;
+							let init_lookup = t.as_class()?.env.lookup(
+								&Symbol {
+									name: if t.is_preflight_class() {
+										CLASS_INIT_NAME.into()
+									} else {
+										CLASS_INFLIGHT_INIT_NAME.into()
+									},
+									span: Default::default(),
+								},
+								None,
+							);
```

Comment:
Can use `into()` to automatically create a Symbol here

```suggestion
							let init_lookup = t.as_class()?.env.lookup(
								&if t.is_preflight_class() {
										CLASS_INIT_NAME
									} else {
										CLASS_INFLIGHT_INIT_NAME
									}.into(),
								None,
							);
```